src/libraries/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.Unix.cs

```diff
@@ -7,6 +7,13 @@ public partial class ZipArchiveEntry
     {
         internal const ZipVersionMadeByPlatform CurrentZipPlatform = ZipVersionMadeByPlatform.Unix;
 
+        /// <summary>
+        /// The default entry permissions are used to support zip archives on multiple platforms.
+        /// For Unix the default permissions are an equivalent of (Convert.ToInt32("644", 8) &lt;&lt; 16)
+        /// where 644 are RW-R-R permissions.
+        /// </summary>
+        internal const int DefaultEntryPermissions = 420 << 16;
```

Comment:
the `<< 16` is to match where they're stored in ExternalAttributes, right? The perms are actually 420==0644 so the name of this is confusing to me. Can the shift happen at the point its set on ExternalAttributes?

Reply:
Or, since you are just going to set them not or them, you could call this `DefaultExternalAttributes`

Reply:
I agree, I renamed the const to be `DefaultExternalAttributes` as you suggested.

Reply:
This has since then been renamed back as now I am using the enum in the constants and I am shifting it right where we set the attributes.